[BUG] Remove check_X_y for pipelines

RikStarmans opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look
at our contribution guide: https://github.com/sktime/sktime/blob/main/CONTRIBUTING.md
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.

Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->
Fixes https://github.com/sktime/sktime/issues/3776

#### What does this implement/fix? Explain your changes.
<!--
A clear and concise description of what you have implemented.
-->
A forecaster which contains a transformer for both X and Y and converts series to panel, cannot be used.
The following forecaster does not make sense. Operations like these are needed for temporal aggregators.
```python
from sktime.datasets import load_airline
from sktime.forecasting.naive import NaiveForecaster
from sktime.transformations.bootstrap import STLBootstrapTransformer
y = load_airline()
f = STLBootstrapTransformer()*(STLBootstrapTransformer()**NaiveForecaster())
f.fit(y=y)
# fail in the current code base, due to TypeError: X and y must have the same scitype
# it should not fail as X and y have the same type after the trafos
# this is fixed by this push
f.fit(y=y,
      X=y)
# the following still results in a TypeError
f = NaiveForecaster()
try:
    f.fit(y=y,
          X=STLBootstrapTransformer().fit_transform(y))
except TypeError:
    print('Catched a type error as expected')
```

#### Does your contribution introduce a new dependency? If yes, which one?
No
<!--
If your contribution does add a new hard dependency, we may suggest to initially develop your contribution in a separate companion package in https://github.com/sktime/ to keep external dependencies of the core sktime package to a minimum.
-->

#### What should a reviewer concentrate their feedback on?
Changes are minimal, I did not do any test or added any tests.
<!-- This section is particularly useful if you have a pull request that is still in development. You can guide the reviews to focus on the parts that are ready for their comments. We suggest using bullets (indicated by * or -) and filled checkboxes [x] here -->

#### Any other comments?
<!--
Please be aware that we are a loose team of volunteers so patience is necessary; assistance handling other issues is very welcome. We value all user contributions, no matter how minor they are. If we are slow to review, either the pull request needs some benchmarking, tinkering, convincing, etc. or more likely the reviewers are simply busy. In either case, we ask for your understanding during the review process.
-->

#### PR checklist
<!--
Please go through the checklist below. Please feel free to remove points if they are not applicable.
-->

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/sktime/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.
- [ x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.


<!--
Thanks for contributing!
-->

fkiraly reviewed (changes requested):
Great, thanks for the initiative!

May I request:

* I would suggest a slightly more principled version of this along the lines in my newest post in #3776, here: https://github.com/sktime/sktime/issues/3776#issuecomment-1315642986
* also, kindly make sure you don't accidentally include your other pull request (is that intended or accident? If intended, comment on this in the PR description, can make sense for testing or conditionalities)

RikStarmans commented:
> Great, thanks for the initiative!
> 
> May I request:
> 
> * I would suggest a slightly more principled version of this along the lines in my newest post in #3776, here: [#3776 (comment)](https://github.com/sktime/sktime/issues/3776#issuecomment-1315642986)
Fixed
> * also, kindly make sure you don't accidentally include your other pull request (is that intended or accident? If intended, comment on this in the PR description, can make sense for testing or conditionalities)
Fixed

fkiraly reviewed (changes requested):
Yes, makes sense.

* linting fails, should be fixed
* should the check not be in the line `if y_scitype != X_scitype:`?
